js/libtextsecure.js

```diff
@@ -37666,8 +37666,22 @@ var TextSecureServer = (function() {
         return true;
     }
 
+    // On Linux/Electron multiple quick web requests can result in the Node.js event
+    //   loop getting wedged. Bug: https://github.com/electron/electron/issues/10570
+    //   This forces the event loop to move.
+    function scheduleHangWorkaround() {
+        setTimeout(function() {
+            setImmediate(function() {
```

Comment:
Does it work if you just call setImmediate (not wrapped in a setTimeout) ?

Reply:
I'm very sure that calling this method fixes things when they are wedged. Calling it before the request which causes the wedged state might work, but I'm not really interested in digging in to try to test that out.

From a theory perspective, I expect that this works because it forces the event loop to start up, so we should delay until there's a potential stopped event loop.